stan/math/opencl/rev/tanh.hpp

```diff
@@ -0,0 +1,36 @@
+#ifndef STAN_MATH_OPENCL_REV_TANH_HPP
+#define STAN_MATH_OPENCL_REV_TANH_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/kernel_generator.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/math/rev/fun/value_of.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns the elementwise `tanh()` of a var_value<matrix_cl<double>>
+ * in radians.
+ *
+ * @param A argument
+ * @return Elementwise `tanh()` of the input, in radians.
+ */
+inline var_value<matrix_cl<double>> tanh(
+    const var_value<matrix_cl<double>>& A) {
+  var_value<matrix_cl<double>> res = tanh(A.val());
+
+  reverse_pass_callback([A, res]() mutable {
+    const matrix_cl<double>& cosh_A_val = cosh(A.val());
```

Comment:
```suggestion
    auto cosh_A_val = cosh(A.val());
```
No need to eval it in a separate kernel. Kernel generator will deduplicate iit, so cosh will only be calculated once.